cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateTestExecutableTask.java

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package io.ballerina.cli.task;
+
+import io.ballerina.cli.utils.BuildTime;
+import io.ballerina.cli.utils.TestUtils;
+import io.ballerina.projects.EmitResult;
+import io.ballerina.projects.JBallerinaBackend;
+import io.ballerina.projects.JarLibrary;
+import io.ballerina.projects.JarResolver;
+import io.ballerina.projects.JvmTarget;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.ModuleDescriptor;
+import io.ballerina.projects.PackageCompilation;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ProjectException;
+import io.ballerina.projects.internal.model.Target;
+import io.ballerina.projects.util.ProjectConstants;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.ballerinalang.test.runtime.entity.TestSuite;
+import org.ballerinalang.testerina.core.TestProcessor;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.zip.ZipFile;
+
+import static io.ballerina.cli.launcher.LauncherUtils.createLauncherException;
+import static io.ballerina.projects.util.ProjectConstants.USER_DIR;
+
+public class CreateTestExecutableTask extends CreateExecutableTask {
```

Comment:
What's the benefit of implementing the `CreateExecutableTask`?
IMO it's better to stick to the existing semantics of the tasks and implement `Task` interface instead.

Reply:
The benefits would be the similarities between these two classes which allowed reusing some methods and properties via inheritance. Should I move back to Task implementation? In that case, I would have to write some redundant code.

Reply:
Does it make logical sense for the `CreateTestExecutableTask` to inherit the `CreateExecutableTask`? AFAIU these are two classes at the same level which do similar tasks. If you want to reuse the code, perhaps we can have an abstract class/ interface that is implemented by both these classes.

Reply:
Other than the re usability, there are no other reasons. 

> Does it make logical sense for the `CreateTestExecutableTask` to inherit the `CreateExecutableTask`? AFAIU these are two classes at the same level which do similar tasks. If you want to reuse the code, perhaps we can have an abstract class/ interface that is implemented by both these classes.

That seems to be the case, I only thought from the perspective of re usability. Upon checking the code, there's one function which I moved to an utility class and other functions which I redefined in the class again. I changed the class implementation to match the structure. These new changes can be found in 73fb681d399dd985d8d8bb8560fa216543b6b028 commit. Still believe there can be more improvements such as an interface like you've mentioned.